astropy/coordinates/sky_coordinate.py

```diff
@@ -377,8 +377,16 @@ def __eq__(self, value):
         equivalent, extra frame attributes are equivalent, and that the
         representation data are exactly equal.
         """
+
+        if isinstance(value, BaseCoordinateFrame):
+            if value._data is None:
+                raise ValueError("Can only compare SkyCoord to Frame with data")
```

Comment:
This case should be covered with a test.

Reply:
done